library/ix-dev/community/rust-desk/upgrade_strategy

```diff
@@ -0,0 +1,31 @@
+#!/usr/bin/python3
+import json
+import re
+import sys
+
+from catalog_update.upgrade_strategy import semantic_versioning
+
+
+RE_STABLE_VERSION = re.compile(r'\d+\.\d+\.\d+(-\d+)?')
+
+
+def newer_mapping(image_tags):
+    key = list(image_tags.keys())[0]
+    tags = {t: t for t in image_tags[key] if RE_STABLE_VERSION.fullmatch(t)}
+    version = semantic_versioning(list(tags))
```

Comment:
Curious if semantic versioning works fine with the hyphen involved?

Reply:
Updated the strategy and now, 

With input
```
{
  "image": [
      "1.1.0",
      "1.1.0-2",
      "1.1.0-3"
  ]
}
```

I get output

```
{"tags": {"image": "1.1.0-3"}, "app_version": "1.1.0.3"}
```